pinot-spi/src/main/java/org/apache/pinot/spi/annotations/ScalarFunction.java

```diff
@@ -49,4 +49,7 @@
   // If empty, FunctionsRegistry registers the method name as function name;
   // If not empty, FunctionsRegistry only registers the function names specified here, the method name is ignored.
   String[] names() default {};
+
+  // Whether the scalar function expects and can handle null arguments.
+  boolean nullableParameters() default false;
```

Comment:
rename to allowNullParameters since it's a boolean.

Reply:
@kishoreg Will address in a separate PR. fyi: @Jackie-Jiang